ast-nodes/index-access.dart

```diff
@@ -0,0 +1,24 @@
+import 'modifiable-primary.dart';
+import 'expression.dart';
+
+/// An array element access by [index] – for either reading or writing.
+///
+/// Chained element access requires several [IndexAccess] objects:
+/// ```dart
+/// // "a[0][1]" is represented with
+/// IndexAccess(
+///   IntegerLiteral(1),
+///   IndexAccess(
+///     IntegerLiteral(0),
+///     Variable("a"),
+///   ),
+/// )
+/// ```
+class IndexAccess implements ModifiablePrimary {
+  Expression index;
```

Comment:
Shouldn't it be of type Sum? Otherwise, we could allow accessing array elements using booleans but is supposed to be Ints.

Reply:
Type checking is out of scope of the syntax analyzer, it should allow anything